src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -37,11 +37,12 @@ mod move_error;
 
 pub fn gather_loans_in_fn<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
                                     body: hir::BodyId)
-                                    -> (Vec<Loan<'tcx>>, move_data::MoveData<'tcx>) {
+                                    -> (Vec<SafeLoan>, Vec<Loan<'tcx>>, move_data::MoveData<'tcx>) {
```

Comment:
So I feel like the *right* fix here might be removing the notion of `SafeLoan` altogether, and adjusting the follow-on code, per the strategy I describe in the NLL RFC.